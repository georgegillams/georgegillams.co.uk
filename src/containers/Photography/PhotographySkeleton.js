import React from 'react';
import Skeleton from '@george-gillams/components/skeleton';

const PhotographySkeleton = () => <Skeleton style={{ width: '100%', height: '84' }} />;

export default PhotographySkeleton;
