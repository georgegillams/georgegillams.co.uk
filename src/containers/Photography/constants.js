export const KEY = 'photography';
export const LOAD_PHOTOS = `app/${KEY}/LOAD_PHOTOS`;
