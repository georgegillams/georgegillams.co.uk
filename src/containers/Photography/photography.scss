@import '~utils/tokens';

.photography {
  &__card-container {
    max-width: 90rem;

    @include card-container;
  }

  &__card {
    margin: 0 0.5rem $spacing-base 0.5rem;
  }

  &__subsection {
    @include centred;
  }

  &__image {
    width: 70%;
    margin-bottom: $spacing-lg;

    @include bpk-breakpoint-mobile {
      width: 90%;
    }
  }
}
