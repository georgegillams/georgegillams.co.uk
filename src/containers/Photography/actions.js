import { createRoutine } from 'redux-saga-routines';

import { LOAD_PHOTOS } from './constants';

export const loadPhotos = createRoutine(LOAD_PHOTOS);
