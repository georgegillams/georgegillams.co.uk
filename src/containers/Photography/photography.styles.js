import { breakpointMd, spacingBase, spacingLg } from '@george-gillams/components/constants/layout';
import styled from 'styled-components';
import Image from '@george-gillams/components/image';

export const LinkContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: ${spacingLg} 0;

  > *:not(:last-child) {
    margin-bottom: ${spacingBase};
  }
`;

export const ImageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: ${spacingLg};
  margin-bottom: ${spacingLg};

  @media (min-width: ${breakpointMd}) {
    flex-direction: row;
    align-items: center;
  }
`;

export const StyledImage = styled(Image)`
  width: 12rem;

  :nth-child(2) {
    margin-top: ${spacingLg};
  }

  @media (min-width: ${breakpointMd}) {
    width: 14rem;
    margin: 0 ${spacingLg};

    :nth-child(2) {
      margin-top: 0;
      width: 16.3rem;
    }
  }
`;
