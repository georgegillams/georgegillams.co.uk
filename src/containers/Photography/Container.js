import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import PageTitle from 'components/common/PageTitle';

import { CreativeCommons } from 'components/CreativeCommons';
import TextLink from 'components/common/TextLink';
import PageContainer from 'components/common/PageContainer';
import PhotographySkeleton from './PhotographySkeleton';
import LoadingCover from '@george-gillams/components/loading-cover';
import { ImageWrapper, LinkContainer, StyledImage } from './photography.styles';
import PhotoGallery from 'components/photo-gallery';
import { ANIMATIONS, ScrollAnimationWrapper, withScrollAnimation } from '@george-gillams/components/effects';
import Button from 'components/common/Button/Button';
import IopMember from './images/iop-member.jpg';
import GuildQualified from './images/guild-qualified.jpg';

const CreativeCommonsWithScroll = withScrollAnimation(CreativeCommons, { animation: ANIMATIONS.fade });

const Photography = props => {
  const { ssrPhotos, loadPhotos, photographyState } = props;

  useEffect(() => {
    loadPhotos();
  }, []);

  const photosToRender = JSON.parse(JSON.stringify(photographyState.photos || ssrPhotos));

  return (
    <>
      <PageContainer centred bottomPadding>
        <PageTitle anchor={false} name="Photography" />
        <ScrollAnimationWrapper>
          <LinkContainer>
            <Button hrefExternal href="https://www.flickr.com/photos/georgegillams">
              View all my photos on Flickr
            </Button>
            <TextLink hrefExternal href="https://instagram.com/tiggy.and.bamse">
              See Tigger and Bamse&#39;s Instagram üòª üê∂
            </TextLink>
            <TextLink hrefExternal href="https://unsplash.com/@georgegillams">
              View my profile on Unsplash
            </TextLink>
            <TextLink hrefExternal href="https://gurushots.com/georgegillams/achievements">
              Find me on GuruShots
            </TextLink>
            <TextLink href="/contact">Contact me about photography work</TextLink>
          </LinkContainer>
        </ScrollAnimationWrapper>
        <ScrollAnimationWrapper>
          <ImageWrapper>
            <StyledImage
              imgProps={{
                alt: 'I have IOP membership having completed their Professional Diploma in Photography.',
              }}
              aspectX={1589}
              aspectY={850}
              lightSrc={IopMember.src}
              darkSrc={IopMember.src}
            />
            <StyledImage
              imgProps={{
                alt: 'I am a qualified member of the Guild of Photography.',
              }}
              aspectX={1417}
              aspectY={650}
              lightSrc={GuildQualified.src}
              darkSrc={GuildQualified.src}
            />
          </ImageWrapper>
        </ScrollAnimationWrapper>
        <ScrollAnimationWrapper style={{ width: '100%' }}>
          <LoadingCover
            loadingSkeleton={PhotographySkeleton}
            loading={!photosToRender || !photosToRender.length}
            error={!!photographyState.photosLoadError}>
            <>
              <PhotoGallery photos={photosToRender} />
            </>
          </LoadingCover>
        </ScrollAnimationWrapper>
      </PageContainer>
      <CreativeCommonsWithScroll />
    </>
  );
};

Photography.propTypes = {
  photographyState: PropTypes.object,
  ssrPhotos: PropTypes.arrayOf(PropTypes.object),
  loadPhotos: PropTypes.func.isRequired,
};

Photography.defaultProps = { photographyState: {}, ssrPhotos: [] };

export default Photography;
