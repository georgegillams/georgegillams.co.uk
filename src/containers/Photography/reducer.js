import produce from 'immer';

import { loadPhotos } from './actions';

export const initialState = {
  photos: null,
  loadingPhotos: false,
  loadPhotosError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case loadPhotos.REQUEST:
        draft.loadingPhotos = true;
        draft.loadPhotosError = null;
        break;

      case loadPhotos.SUCCESS:
        draft.loadingPhotos = false;
        draft.photos = payload.photos;
        break;

      case loadPhotos.FAILURE:
        draft.loadingPhotos = false;
        draft.loadPhotosError = payload;
        break;
    }
  });

export default reducer;
