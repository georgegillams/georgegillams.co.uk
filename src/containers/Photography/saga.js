import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';

import { loadPhotos } from './actions';

export function* doLoadPhotos() {
  const requestURL = apiStructure.loadShowcaseImages.fullPath;

  try {
    yield put(loadPhotos.request());

    const result = yield call(request, requestURL, {
      method: 'GET',
    });

    if (result.error) {
      yield put(loadPhotos.failure(result));
    } else {
      yield put(loadPhotos.success(result));
    }
  } catch (err) {
    yield put(loadPhotos.failure(err));
  } finally {
    yield put(loadPhotos.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadPhotos.TRIGGER, doLoadPhotos);
}
