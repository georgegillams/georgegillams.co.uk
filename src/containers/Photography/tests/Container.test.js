import React from 'react';
import { render } from '@testing-library/react';

import Photography from '../Container';

describe('<Photography />', () => {
  it('should render correctly', () => {
    const { container } = render(<Photography />);

    expect(container).toMatchSnapshot();
  });
});
