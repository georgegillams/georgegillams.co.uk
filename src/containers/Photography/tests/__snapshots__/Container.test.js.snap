// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photography /> should render correctly 1`] = `
<div>
  <h1
    class="page-title__heading"
  >
    Photography
  </h1>
  <div
    class="photography__card-container"
  >
    <a
      aria-label="Find me on GuruShots"
      class="card feature-card feature-card--narrow-compact photography__card card--atomic card--not-disabled"
      href="https://gurushots.com/georgegillams/achievements"
      tabindex="0"
    >
      <div
        class="card__outer-container"
      >
        <div
          class="card__background"
        />
        <div
          class="card__content-container card__content-container--padded"
        >
          <div
            aria-hidden="true"
            class="feature-card__outer-container feature-card__outer-container--narrow-compact"
          >
            <div
              class="feature-card__date feature-card__date--narrow-compact"
            />
            <div
              class="feature-card__center-container feature-card__center-container--narrow-compact"
            >
              <div
                class="section__outer section__outer--inherit-color feature-card__title"
              >
                <h2
                  class="section__heading section__heading--no-padding"
                  id="find-me-on-gurushots"
                >
                  Find me on GuruShots
                </h2>
              </div>
            </div>
            <div
              class="feature-card__image-container"
            >
              <div
                class="image__outer feature-card__image"
              >
                <div
                  class="image__placeholder"
                  style="padding-bottom: 100%;"
                >
                  <div
                    class="image__spinner-container"
                  >
                    <div
                      class="skeleton__outer image__skeleton image__skeleton--animated"
                    >
                      <div
                        class="skeleton__shimmer"
                        style="margin-left: -0px;"
                      />
                    </div>
                  </div>
                  <img
                    alt="card"
                    class="image__img image__img--animated image__img--hidden image__img--light"
                    src="https://i.imgur.com/Jng7EhH.png"
                  />
                  <img
                    alt="card"
                    class="image__img image__img--animated image__img--hidden image__img--dark"
                    src="https://i.imgur.com/Jng7EhH.png"
                  />
                  <noscript />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </a>
    <a
      aria-label="Find me on Flickr"
      class="card feature-card feature-card--narrow-compact photography__card card--atomic card--not-disabled"
      href="https://www.flickr.com/people/georgegillams/"
      tabindex="0"
    >
      <div
        class="card__outer-container"
      >
        <div
          class="card__background"
        />
        <div
          class="card__content-container card__content-container--padded"
        >
          <div
            aria-hidden="true"
            class="feature-card__outer-container feature-card__outer-container--narrow-compact"
          >
            <div
              class="feature-card__date feature-card__date--narrow-compact"
            />
            <div
              class="feature-card__center-container feature-card__center-container--narrow-compact"
            >
              <div
                class="section__outer section__outer--inherit-color feature-card__title"
              >
                <h2
                  class="section__heading section__heading--no-padding"
                  id="find-me-on-flickr"
                >
                  Find me on Flickr
                </h2>
              </div>
            </div>
            <div
              class="feature-card__image-container"
            >
              <div
                class="image__outer feature-card__image"
              >
                <div
                  class="image__placeholder"
                  style="padding-bottom: 100%;"
                >
                  <div
                    class="image__spinner-container"
                  >
                    <div
                      class="skeleton__outer image__skeleton image__skeleton--animated"
                    >
                      <div
                        class="skeleton__shimmer"
                        style="margin-left: -0px;"
                      />
                    </div>
                  </div>
                  <img
                    alt="card"
                    class="image__img image__img--animated image__img--hidden image__img--light"
                    src="https://i.imgur.com/u30cQWU.png"
                  />
                  <img
                    alt="card"
                    class="image__img image__img--animated image__img--hidden image__img--dark"
                    src="https://i.imgur.com/u30cQWU.png"
                  />
                  <noscript />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </a>
    <a
      aria-label="Find me on Unsplash"
      class="card feature-card feature-card--narrow-compact photography__card card--atomic card--not-disabled"
      href="https://unsplash.com/@georgegillams"
      tabindex="0"
    >
      <div
        class="card__outer-container"
      >
        <div
          class="card__background"
        />
        <div
          class="card__content-container card__content-container--padded"
        >
          <div
            aria-hidden="true"
            class="feature-card__outer-container feature-card__outer-container--narrow-compact"
          >
            <div
              class="feature-card__date feature-card__date--narrow-compact"
            />
            <div
              class="feature-card__center-container feature-card__center-container--narrow-compact"
            >
              <div
                class="section__outer section__outer--inherit-color feature-card__title"
              >
                <h2
                  class="section__heading section__heading--no-padding"
                  id="find-me-on-unsplash"
                >
                  Find me on Unsplash
                </h2>
              </div>
            </div>
            <div
              class="feature-card__image-container"
            >
              <div
                class="image__outer feature-card__image"
              >
                <div
                  class="image__placeholder"
                  style="padding-bottom: 100%;"
                >
                  <div
                    class="image__spinner-container"
                  >
                    <div
                      class="skeleton__outer image__skeleton image__skeleton--animated"
                    >
                      <div
                        class="skeleton__shimmer"
                        style="margin-left: -0px;"
                      />
                    </div>
                  </div>
                  <img
                    alt="card"
                    class="image__img image__img--animated image__img--hidden image__img--light"
                    src="https://i.imgur.com/obBqvqK.png"
                  />
                  <img
                    alt="card"
                    class="image__img image__img--animated image__img--hidden image__img--dark"
                    src="https://i.imgur.com/obBqvqK.png"
                  />
                  <noscript />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="section__outer"
  >
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="harlequins-vs-worcester-rugby-match"
      >
        Harlequins vs Worcester Rugby Match
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 57.96610169491525%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Harlequins vs Worcester Rugby Match"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/OsFI23z.jpg"
          />
          <img
            alt="Harlequins vs Worcester Rugby Match"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/OsFI23z.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="the-cat-looking-handsome-as-ever!"
      >
        The cat looking handsome as ever!
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="The cat looking handsome"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/Aqy3tuA.jpg"
          />
          <img
            alt="The cat looking handsome"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/Aqy3tuA.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="a-training-scenario-at-epicc-2017"
      >
        A training scenario at EPICC 2017
      </h3>
      <div
        class="graphic-content__outer-container photography__image"
      >
        <div
          class="graphic-content__content-container graphic-content__content-container--hidden"
        >
          <div
            class="image__outer"
          >
            <div
              class="image__placeholder"
              style="padding-bottom: 60.416666666666664%;"
            >
              <div
                class="image__spinner-container"
              >
                <div
                  class="skeleton__outer image__skeleton image__skeleton--animated"
                >
                  <div
                    class="skeleton__shimmer"
                    style="margin-left: -0px;"
                  />
                </div>
              </div>
              <img
                alt="A training scenario at EPICC 2017"
                class="image__img image__img--animated image__img--hidden image__img--light"
                src="https://i.imgur.com/6B35GTV.jpg"
              />
              <img
                alt="A training scenario at EPICC 2017"
                class="image__img image__img--animated image__img--hidden image__img--dark"
                src="https://i.imgur.com/6B35GTV.jpg"
              />
              <noscript />
            </div>
          </div>
        </div>
        <div
          class="graphic-content__warning-container"
        >
          <div
            class="graphic-content__warning-container__inner"
          >
            <div
              class="graphic-content__text"
            >
              This image contains graphic content
            </div>
            <br />
            <button
              class="button__outer button__outer--regular button__outer--no-bouncy graphic-content__text"
              type="button"
            >
              <span
                class="button__child"
              >
                Show graphic content
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="dog-running-with-a-stick"
      >
        Dog running with a Stick
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Dog running with a stick"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/8dnCZ5D.jpg"
          />
          <img
            alt="Dog running with a stick"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/8dnCZ5D.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="stunt-motorcyclist"
      >
        Stunt Motorcyclist
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Stunt Motorcyclist standing on the back of a bike"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/WlLYxDw.jpg"
          />
          <img
            alt="Stunt Motorcyclist standing on the back of a bike"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/WlLYxDw.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="longleat-festival-of-light"
      >
        Longleat Festival of Light
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Longleat Festival of Light"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/EHF7zqM.jpg"
          />
          <img
            alt="Longleat Festival of Light"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/EHF7zqM.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="serre-chevalier"
      >
        Serre Chevalier
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Serre Chavelier Ski Slopes"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/SSGGfbL.jpg"
          />
          <img
            alt="Serre Chavelier Ski Slopes"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/SSGGfbL.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="nick-matthew-playing-in-the-canary-wharf-open"
      >
        Nick Matthew playing in the Canary Wharf Open
      </h3>
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Nick Matthew playing in the Canary Wharf Open"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/h4BFWqS.jpg"
          />
          <img
            alt="Nick Matthew playing in the Canary Wharf Open"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/h4BFWqS.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
  </div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="photoshop"
    >
      Photoshop
    </h2>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="tulips-exploding-with-light"
      >
        Tulips exploding with light
      </h3>
      <span
        class="paragraph"
      >
        For some reason the idea of light exploding out of tulips popped into my mind, so I went out to find some and made it a reality. I used a similar effect in Art that I had used in the past to create beams of sunlight breaking through the clouds, and then darkened the background a little.
      </span>
      <br />
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Exploding tulips"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/PIKQ2D6.jpg"
          />
          <img
            alt="Exploding tulips"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/PIKQ2D6.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="miss-saigon-sketch"
      >
        Miss Saigon sketch
      </h3>
      <span
        class="paragraph"
      >
        With Miss Saigon coming to cinemas soon for one day only, I was inspired to draw the production logo (aka tempted to procrastinate).
      </span>
      <br />
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 139.34045517882026%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Miss Saigon sketch"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/y3i2Ll1.jpg"
          />
          <img
            alt="Miss Saigon sketch"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/y3i2Ll1.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="dual-carriageway-light-painting-(in-post)"
      >
        Dual-carriageway light-painting (in post)
      </h3>
      <span
        class="paragraph"
      >
        One evening after leaving work at an unearthly hour, I shot this uninspired photo. In Photoshop, I then used the brush tool and some layer styles to create a 'painting with light' effect.
      </span>
      <br />
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Dual-carriageway light-painting"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/T502lkX.jpg"
          />
          <img
            alt="Dual-carriageway light-painting"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/T502lkX.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
  </div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="food-is-art!-(...sometimes)"
    >
      Food is art! (...sometimes)
    </h2>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="guinness-cake"
      >
        Guinness cake
      </h3>
      <span
        class="paragraph"
      >
        I like spending time on food presentation when the opportunity calls for it. So I created this masterpiece to share with the office and celebrate a legendary drink!
      </span>
      <br />
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Guinness cake"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/oBWlSDO.jpg"
          />
          <img
            alt="Guinness cake"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/oBWlSDO.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
    <div
      class="subsection__outer photography__subsection"
    >
      <h3
        class="subsection__heading"
        id="ratatouille"
      >
        Ratatouille
      </h3>
      <span
        class="paragraph"
      >
        To get us in the mood for Disneyland, I cooked up this Ratatouille in the style of the dish served in the film. It came out better than I expected... Pretty pleased with the result!
      </span>
      <br />
      <div
        class="image__outer photography__image"
      >
        <div
          class="image__placeholder"
          style="padding-bottom: 66.66666666666667%;"
        >
          <div
            class="image__spinner-container"
          >
            <div
              class="skeleton__outer image__skeleton image__skeleton--animated"
            >
              <div
                class="skeleton__shimmer"
                style="margin-left: -0px;"
              />
            </div>
          </div>
          <img
            alt="Ratatouille"
            class="image__img image__img--animated image__img--hidden image__img--light"
            src="https://i.imgur.com/kRGhYxz.jpg"
          />
          <img
            alt="Ratatouille"
            class="image__img image__img--animated image__img--hidden image__img--dark"
            src="https://i.imgur.com/kRGhYxz.jpg"
          />
          <noscript />
        </div>
      </div>
    </div>
  </div>
  <div
    class="creative-commons__wrapper"
  >
    <div
      class="creative-commons__background"
    />
    <div
      class="subsection__outer creative-commons__inner"
    >
      <h3
        class="subsection__heading"
        id="copyright"
      >
        Copyright
      </h3>
      <span
        class="paragraph"
      >
        Most of my photos are licensed under
         
        <a
          aria-label="Creative Commons BY-SA 3.0 (opens in a new tab)"
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="https://creativecommons.org/licenses/by-sa/3.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Creative Commons BY-SA 3.0
          <div
            class="text-link__icon-container"
          >
            <svg
              class="icon text-link__icon"
              height="18"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
              />
            </svg>
          </div>
        </a>
        .
        <br />
        If you are unsure about your right to use them please 
        <a
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="/contact"
        >
          contact me
        </a>
        .
      </span>
    </div>
  </div>
</div>
`;
